### ISSUE TITLE
Directory created in recycle bin does not respect Permissions settings

### ISSUE THREAD

**Issue by froggler** [Radarr/Radarr#10450]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

When a movie is moved to the Recycling Bin (under Settings/Media Management/File Management), the media files are put into a new direcotry created under that recycling bin path. The new directory is created with default permissions, instead of following the "chmod Folder" setting (in Settings/Media Management/Permissions).

For some configurations this leads to files in the recycle bin being unusable, e.g. over network file share where permissions cannot be fixed.

### Expected Behavior

Any folder being created by Radarr should follow the permissions "chmod Folder" setting if enabled. Or if there is an intended separation of behaviour for media library vs recycle bin, there should be an option to adjust this (but that seems like a very niche case).

### Steps To Reproduce

1. With configuration settings for Media Management:
* File Management / Recycling bin: /recyclingbin
* Permissions / Set Permissions: enabled
* Permissions / chmod Folder: 777
2. Import a media file, e.g. MovieName (2024).mov
* Radarr creates "/movies/MovieName (2024)" with permissions 777 per the setting, and puts the movie file(s) in this directory
3. Import a replacement media file for MovieName (2024)
* Radarr will create a new directory: "/recyclingbin/MovieName (2024)" with default permissions (not 777)
* Radarr will move the old MovieName (2024).mov to this new directory
* Radarr will import the new file(s) into /movies/MovieName (2024)
4. When browsing to /recyclingbin the user will discover the permissions are incorrect on "/recyclingbin/MovieName (2024)"

### Environment

```markdown
- OS: Unraid 6.12.10
- Radarr: Radarr 5.9.1.9070 / 5.9.1.9070-ls236 by linuxserver.io
- Docker Install: yes
- Using Reverse Proxy: no
- Browser: N/A
- Database: Sqlite 3.45.3
```

### What branch are you running?

Master

### Trace Logs? **Not Optional**

[radarr.trace.excerpt.txt](https://github.com/user-attachments/files/17059149/radarr.trace.excerpt.txt) - excerpt from the upgrade event

not strictly a log, but ls command on the movie folder + recycled movie folder result in different permissions shown:
drwxrwxrwx 1 nobody users 140 Sep 19 22:35 movies/Pinocchio\ (1940)//
drwxr-xr-x 1 nobody users 0 Sep 19 22:35 recycle/Pinocchio\ (1940)/

### Trace Logs have been provided as applicable. Reports will be closed if the required logs are not provided.

- [X] I have read and followed the steps in the wiki link above and provided the required trace logs - the logs contain `trace` - that are relevant and show this issue.

**Comment by bakerboy448** [comment]
- 777ing permissions is never a sane solution, that's removing all doors and keys because you can't be bothered to turn a knob. Discord for support for that
- If other users accessing your NAS cannot access the folders then fix their access, removing all permissions is not rationale solution.
- change the UMASK of radarr's container to run with open perms 000 which should create with 777/666
- Media management settings apply to import
- Defer to @mynameisbogdan if this should go upstream and if recycle bin should respect source perms and/or media management advanced group/perms settings

**Comment by froggler** [comment]
Geez, I really don't understand the need for patronising rude tone here. I typed up a bug report with example steps that shows the behaviour. Fine, sorry, I'll be more specific, I want 770 on the folders to match the share permissions management in place, which relies on groups, but I'm getting 750. I hope this is acceptable.

UMASK on the container is a reasonable work around - thanks for the suggestion, I'd not thought of this.

My view, if it's worth anything, is the recycling bin is media being managed. It's the same media files, same naming and folder structure, and radarr is the one putting it there (and potentially deleting it on a later schedule). If the user wants a particular set of permissions on the media, I'd think it would apply there too. Especially as it appears, from a user's POV, that the whole containing folder for the movie is being recycled, since the result is an identical folder with the movie's file(s).

**Comment by froggler** [comment]
Yeah they’re on the same volume. Just to clarify, it’s successfully moving the files, but is creating a _new folder_, and its the new folder that doesn’t get the usual permissions applied from settings. I assume the choice to move files not folders is a sensible & intentional one, since the folder could contain unmanaged files.

My interpretation from looking at the code is that there’s a different approach between `MovieFileMovingService` vs `RecycleBinProvider` in how to make a new folder via `DiskProvider.CreateFolder()`:
- `MovieFileMovingService` provides its own `CreateFolder()` that handles permissions after creating the folder via `DiskProvider` (among other things)
- `UpgradeMediaFileService` uses `RecycleBinProvider` for the file operations, which does not set any permissions for newly created folders

The call stack (as best I can make it from reading the code) when Radarr moves an upgraded media file to the recycling bin goes like this:
- starting at `MediaFiles.UpgradeMediaFileService.UpgradeMovieFile()`
- on line 63 `UpgradeMovieFile()` calls `RecycleBinProvider.DeleteFile()` (and as noted above this seems the sensible choice, compared to `DeleteFolder()`)
- on line 97 `DeleteFile()` calls `_diskProvider.CreateFolder(destinationFolder)`
- `NzbDrone.Common.Disk.DiskProviderBase.CreateFolder()` calls `Directory.CreateDirectory()`
  - Note that `NzbCore.MediaFiles.MovieFileMovingService.CreateFolder()` makes a call to `_mediaFileAttributeService.SetFolderPermissions()` when it creates a new directory
  - but neither `RecycleBinProvider.DeleteFile()` nor `DiskProvider.CreateFolder()` here make an equivalent call to `SetFolderPermissions()`
- then back up to `DeleteFile()` we end up at line 122 `_diskTransferService.TransferFile(path, destination, TransferMode.Move)` which moves the file itself

(I could be wrong in exactly which CreateFolder() is being called as I’m not familiar enough with the code to know if there’s any other version of DiskProvider.CreateFolder() to check here.)

Assuming there’s no harm/complexity in exposing the `IMediaFileAttributeService` to `RecycleBinProvider` it would seem to then just require the addition of `_mediaFileAttributeService.SetFolderPermissions()` after line 97 in RecycleBinProvider.cs. I don’t know C# well enough to be certain of this assumption though.
